tex/gregoriotex-main.tex

```diff
@@ -405,86 +402,81 @@
+    \else%%% DEPRECATED for removal in 5.0
+      \gre@style@initial% keep this line
+      \global\gre@dimen@temp@four = \gre@dimen@beforeinitialshift\relax% keep this line
+      \endgre@style@initial% keep this line
+    \fi%%% DEPRECATED for removal in 5.0
+    \ifx\gre@empty@biginitialformat\grebiginitialformat% OBSOLETE
+    \else%  OBSOLETE
+      \gre@obsolete{\protect\grebiginitialformat}{\protect\grechangestyle{biginitial}}%  OBSOLETE
+    \fi%  OBSOLETE
   \fi %
   \hskip\gre@dimen@temp@four %
   \global\advance\gre@dimen@initialwidth by \gre@dimen@temp@four %
   \gre@debugmsg{annotation}{Ready to place initial.}%
   \box\gre@box@initial%
   \ifnum\gre@biginitial=0\relax%
-    \ifx\gre@empty@initialformat\greinitialformat% DEPRECATED
+    \gre@style@initial%
+    \global\gre@dimen@temp@four = \gre@dimen@afterinitialshift\relax%
+    \endgre@style@initial%
+    \ifx\gre@empty@initialformat\greinitialformat% OBSOLETE
+    \else%  OBSOLETE
+      \gre@obsolete{\protect\greinitialformat}{\protect\grechangestyle{initial}}%  OBSOLETE
+    \fi%  OBSOLETE
+  \else%
+    \ifgre@allowdeprecated%%% DEPRECATED for removal in 5.0
+      \gre@style@biginitial%%% DEPRECATED for removal in 5.0
+      \global\gre@dimen@temp@four = \gre@dimen@afterinitialshift\relax%%% DEPRECATED for removal in 5.0
+      \endgre@style@biginitial%%% DEPRECATED for removal in 5.0
+    \else%%% DEPRECATED for removal in 5.0
       \gre@style@initial% keep this line
       \global\gre@dimen@temp@four = \gre@dimen@afterinitialshift\relax% keep this line
       \endgre@style@initial% keep this line
-    \else%  DEPRECATED
-      \gre@deprecated{\protect\greinitialformat}{\protect\grechangestyle{initial}}%  DEPRECATED
-      \greinitialformat{\global\gre@dimen@temp@four = \gre@dimen@afterinitialshift}%  DEPRECATED
-    \fi%  DEPRECATED
-  \else%
-    \ifx\gre@empty@biginitialformat\grebiginitialformat% DEPRECATED
-      \ifgre@allowdeprecated%%% DEPRECATED for removal in 5.0
-        \gre@style@biginitial%%% DEPRECATED for removal in 5.0
-        \global\gre@dimen@temp@four = \gre@dimen@afterinitialshift\relax%%% DEPRECATED for removal in 5.0
-        \endgre@style@biginitial%%% DEPRECATED for removal in 5.0
-      \else%%% DEPRECATED for removal in 5.0
-        \gre@style@initial% keep this line
-        \global\gre@dimen@temp@four = \gre@dimen@afterinitialshift\relax% keep this line
-        \endgre@style@initial% keep this line
-      \fi%%% DEPRECATED for removal in 5.0
-    \else%  DEPRECATED
-      \gre@deprecated{\protect\grebiginitialformat}{\protect\grechangestyle{biginitial}}%  DEPRECATED
-      \grebiginitialformat{\global\gre@dimen@temp@four = \gre@dimen@afterinitialshift}%  DEPRECATED
-    \fi%  DEPRECATED
+    \fi%%% DEPRECATED for removal in 5.0
+    \ifx\gre@empty@biginitialformat\grebiginitialformat% OBSOLETE
+    \else%  OBSOLETE
+      \gre@obsolete{\protect\grebiginitialformat}{\protect\grechangestyle{biginitial}}%  OBSOLETE
```

Comment:
Along those lines, do you think it would be a good idea to add another class of tests which specifically checks for backwards compatibility with deprecated functions?  I know I changed all the current tests to prohibit the use of deprectated functions, but a separate class (where tests go to die) might help make sure we keep things backwards compatible.